ftw/linkchecker/linkchecker.py

```diff
@@ -0,0 +1,100 @@
+from multiprocessing import Pool
+import time
+import requests
+import urls_testing
+import os
+import sys
+
+urls = urls_testing.urls
+number_of_processes = len(urls)
+timeout = 1
+
+
+def millis():
+    return int(round(time.time() * 1000))
+
+
+def get_uri_responses(urls):
+    headers = {
+        'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.102 Safari/537.36'
+    }
+    error = None
+    response = None
+    start_time = millis()
+    try:
+        response = requests.head(urls.get('destination'),
+                                 timeout=timeout,
+                                 headers=headers,
+                                 allow_redirects=False)
+    except requests.exceptions.Timeout:
+        error = 'Timeout'
+    except requests.exceptions.TooManyRedirects:
+        error = 'Too many redirects'
+    except requests.exceptions.ConnectionError:
+        error = 'Connection Error'
+    except requests.exceptions.RequestException as e:
+        error = e
+    time = millis() - start_time
+    result = {
+        'origin': urls.get('origin'),
+        'destination': urls.get('destination'),
+        'data': response,
```

Comment:
I would not return the whole response but extract the important data already here. This will make the code below much simpler.

Reply:
How would it make it simpler? Wouldn't I still have to do exactly the same, but move it into my worker-function? And is it really a good idea to relocate more than necessary into the multiprocessor worker-function?

Reply:
you lost me here.

Reply:
I adapted this into more readable code.